### PR TITLE
Reorganize lazy creation of MBeanServer

### PR DESCRIPTION
**Pull request by keithc-ca**
A call to `MBeanServerFactory.createMBeanServer()` can fail; thus it is not reasonable to include such a call in the static initializer of a class. This reorganizes that call into a method where an exception can properly be propagate back to the caller of `ManagementFactory.getPlatformMBeanServer()`.

Fixes: #6623
Signed-off-by: Keith W. Campbell <keithc@ca.ibm.com>

### CODE REVIEW

File: jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java [hunk 1]
```diff
@@ -466,51 +466,55 @@ public static ThreadMXBean getThreadMXBean() {
 		return ManagementUtils.getPlatformManagementInterfaces();
 	}
 
-	private static final class ServerHolder {
+	private static final class ServerHolder implements PrivilegedAction<MBeanServer> {
```
Comment (DanHeidinga) [inline jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java]: Can you document this class?

File: jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java [hunk 2]
```diff
@@ -466,51 +466,55 @@ public static ThreadMXBean getThreadMXBean() {
...
 
-		static final MBeanServer platformServer;
+		private static final ServerHolder instance = new ServerHolder();
 
-		static {
-			platformServer = MBeanServerFactory.createMBeanServer();
+		private MBeanServer platformServer;
```
Comment (DanHeidinga) [inline jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java]: Shouldn't this be `volatile` to ensure coherence?

Comment (DanHeidinga) [reply jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java]: Alternatively, using an enum to create the singleton is the approach advocated in Effective Java

Comment (keithc-ca) [reply jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java]: Initialization is synchronized so volatile is not needed.
I believe using an enum would yield the same behavior this is intended to fix.

File: jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java (hunk 1 above)
Comment (keithc-ca) [reply jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java]: Sure.

File: jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java (hunk 2 above)
Comment (DanHeidinga) [reply jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java]: Unless its volatile, there is no guaranteed happens-before relationship between allocation and assignment (unless there are final fields in the object).

Comment (keithc-ca) [reply jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java]: I will make that field `volatile`.

**Comment by keithc-ca** [comment]
Made requested changes and rebased.

**Comment by DanHeidinga** [comment]
Jenkins test sanity osx jdk11

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk8

**Pull request merged**